### PR TITLE
3 style the homepage

### PR DESCRIPTION
**Pull request by AgneSz**

### CODE REVIEW

**Comment by AgneSz** [comment]
Screenshot of index page in the browser:
Screenshot 2022-04-27 at 19 20 17

File: smartflix/app/controllers/static_pages_controller.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
+require 'csv'
+
 class StaticPagesController < ApplicationController
-  def index
-    render html: "smartfix"
+  def home
+    # @movies = CSV.foreach(("db/netflix_titles.csv"), headers: true, col_sep: ",")
```
Comment (terenceponce) [inline smartflix/app/controllers/static_pages_controller.rb:5]: Very minor, but is leaving these comments intentional?

**Review by terenceponce — COMMENTED** [review]
The PR looks good, but I'm curious about a few things:

1. There were a couple of instances where commented code was checked into the commit. I wonder if you were specifically instructed to leave these in. The auto-generated comments are fine to be checked in because those are usually intentional.
2. You specifically added HAML to this commit, but some of the added pages are in ERB, so not everything is in HAML.
3. Some of the files that were added either have weird indentation or does not use Unix-style line endings. Perhaps using a linter like [rubocop](https://github.com/rubocop/rubocop) would be useful for automatically handling these things for you?

Again, the PR itself is fine. The things I've pointed out are mostly fluff, so there are no issues with merging the PR as is.

File: smartflix/app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+small, strike, strong, sub, sup, tt, var,
+b, u, i, center,
+dl, dt, dd, ol, ul, li,
+fieldset, form, label, legend,
+table, caption, tbody, tfoot, thead, tr, th, td,
+article, aside, canvas, details, embed, 
+figure, figcaption, footer, header, hgroup, 
+menu, nav, output, ruby, section, summary,
+time, mark, audio, video {
+	margin: 0;
```
Comment (Swiftda01) [inline smartflix/app/assets/stylesheets/application.scss:15]: V minor thing but looks like this is using tab size 4, I believe the CSS convention is to use 1 tab (2 spaces) for indentation

File: smartflix/app/views/movies/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<div>
+<%= link_to "Previous Page", root_path(page: @page_number - 1) %>
+<%= link_to "Next Page", root_path(page: @page_number + 1) %>
+<h6>Current page number: <%= @page_number %></h6>
```
Comment (Swiftda01) [inline smartflix/app/views/movies/index.html.erb:4]: It's good to get into the habit of using CSS opposed to header tags to determine the size of contents on the page, as header tags have semantic meaning as well as just inbuilt styling. Instead you could add a class to your elements (see below) and then style those elements accordingly. The semantics of HTML are important for assistive devices such as screen readers that use HTML tags to determine what an element represents.
```suggestion
<span class="page-number">Current page number: <%= @page_number %></span>
```

File: smartflix/spec/features/movies_spec.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require_relative '../rails_helper'
+
+RSpec.feature "User visits home page", :type => :feature do
+  it "Displays movies details" do
+    visit "/"
+
+      expect(page).to have_text("Movie titles")
+      expect(page).to have_text("Dick Johnson Is Dead")
+      expect(page).to have_text("Kirsten Johnson")
+      expect(page).to have_text("As her father nears the end of his life")
```
Comment (Swiftda01) [inline smartflix/spec/features/movies_spec.rb:10]: Indentation looks a little off here

File: smartflix/app/controllers/movies_controller.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  def index
+    @movies = CSV.read(("db/netflix_titles.csv"), headers: true, col_sep: ",")
+    @page_number = params.fetch(:page, 1).to_i
+    @page_number = @page_number <= 0? 1 : @page_number
+    @movies_per_page = 50
+    @total_page_number = @movies.length / @movies_per_page
+    # @movies = @movies.take(@movies_per_page)
+    # @movies = @movies.values_at(@movies_per_page*0..@movies_per_page*1-1)
+    # @movies = @movies.values_at(@movies_per_page*1..@movies_per_page*2-1)
+    # @movies = @movies.values_at(@movies_per_page*2..@movies_per_page*3-1)
```
Comment (SimonChenboYi) [inline smartflix/app/controllers/movies_controller.rb:13]: please remove above commented out code.

File: smartflix/app/assets/stylesheets/application.scss (hunk 2 above)
Comment (AgneSz) [reply smartflix/app/assets/stylesheets/application.scss:15]: Changed to Tab Size: 1 (it was set up to 2)

File: smartflix/app/controllers/static_pages_controller.rb (hunk 1 above)
Comment (AgneSz) [reply smartflix/app/controllers/static_pages_controller.rb]: This controller has now been removed as it has no more use in the app

File: smartflix/app/controllers/movies_controller.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+require 'csv'
+
+class MoviesController < ApplicationController
+  def index
+    @movies = CSV.read(("db/netflix_titles.csv"), headers: true, col_sep: ",")
+    @page_number = params.fetch(:page, 1).to_i
+    @page_number = @page_number <= 0? 1 : @page_number
+    @movies_per_page = 50
```
[the remark this answers is not in the document]
Comment (AgneSz) [reply smartflix/app/controllers/movies_controller.rb]: Changed now to constant

File: smartflix/app/controllers/movies_controller.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+require 'csv'
+
```
[the remark this answers is not in the document]
Comment (AgneSz) [reply smartflix/app/controllers/movies_controller.rb]: Now added

File: smartflix/app/controllers/movies_controller.rb [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    @page_number = params.fetch(:page, 1).to_i
+    @page_number = @page_number <= 0? 1 : @page_number
+    @movies_per_page = 50
+    @total_page_number = @movies.length / @movies_per_page
+    # @movies = @movies.take(@movies_per_page)
+    # @movies = @movies.values_at(@movies_per_page*0..@movies_per_page*1-1)
+    # @movies = @movies.values_at(@movies_per_page*1..@movies_per_page*2-1)
+    # @movies = @movies.values_at(@movies_per_page*2..@movies_per_page*3-1)
+
+    @movies = @movies.values_at(@movies_per_page*(@page_number-1)..@movies_per_page*@page_number-1)
```
[the remark this answers is not in the document]
Comment (AgneSz) [reply smartflix/app/controllers/movies_controller.rb]: This is no longer used in index method and has been replaced by function in paginator.rb file

File: smartflix/app/controllers/movies_controller.rb [hunk 9]
```diff
@@ -0,0 +1,17 @@
+require 'csv'
+
+class MoviesController < ApplicationController
+  def index
+    @movies = CSV.read(("db/netflix_titles.csv"), headers: true, col_sep: ",")
```
[the remark this answers is not in the document]
Comment (AgneSz) [reply smartflix/app/controllers/movies_controller.rb]: Index method has now been replaced by function in paginator.rb file

**Pull request merged**